ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -130,13 +130,14 @@ sealed abstract class Queries(tablePrefix: String) {
      """,
   )
 
-  protected[this] def initDatabaseDdls: Vector[InitDdl] =
+  private[this] def initDatabaseDdls: Vector[InitDdl] =
     Vector(
       createTemplateIdsTable,
       createOffsetTable,
       createContractsTable,
-      createVersionTable,
-    )
+    ) ++ extraDatabaseDdls :+ createVersionTable
+
+  protected[this] def extraDatabaseDdls: Seq[InitDdl]
```

Comment:
I will remember that approach, that one is much more waterproofed than what I did before to ensure that the version table is created last.

Reply:
Honestly it was a good excuse to finally do it right, it's been bugging me for a while, even though I added it in the first place. `super` is always a hack.